lib/engine/game/g_18_cz/step/buy_train.rb

```diff
@@ -74,6 +74,18 @@ def train_available?(entity, train)
           end
 
           def check_for_cheapest_train(entity, train); end
+
+          def cheapest_train_price(corporation)
+            cheapest_train = @depot.min_depot_train.variants.values.find do |item|
+              @game.train_of_size?(item, corporation.type)
+            end
+            cheapest_train[:price]
+          end
+
+          def must_take_loan?(corporation)
+            price = cheapest_train_price(corporation)
+            @game.buying_power(corporation) < price
```

Comment:
This shows the loan message even when the president has enough cash

![Screenshot 2021-03-12 212403](https://user-images.githubusercontent.com/1045173/111018935-4f523080-8379-11eb-8cf6-6ab5cc411f6a.png)

gist - https://gist.github.com/michaeljb/bcb8e0a1ad8644ae0d3c32e8301baa40


Looks like this fixes it:

```suggestion
            (@game.buying_power(corporation) + corporation.owner.cash) < price
```

Reply:
Once this is fixed, I think the PR is good to go 👍

Reply:
thanks for double checking :)